### PR TITLE
[4.0][migrator] Do proper lookup and avoid prefixing typeof(of:) if the only found name is the stdlib one.

### PR DESCRIPTION
**Pull request by akyrtzi**
master: https://github.com/apple/swift/pull/9688

rdar://32229317

### CODE REVIEW

**Comment by akyrtzi** [comment]
@swift-ci Please test

**Comment by akyrtzi** [comment]
Explanation: Migrator pass adds 'Swift.' prefix for 'type(of:)' uses unnecessarily. It should onlt add it when lookup for 'type' at that position finds something other than the stdlib function.
Scope: Affects a migrator pass
Radar: rdar://32229317
Risk: Very low, changes only affect a migrator pass

**Comment by akyrtzi** [comment]
@swift-ci Please test macOS

File: lib/Migrator/TypeOfMigratorPass.cpp [hunk 1]
```diff
@@ -39,12 +41,22 @@ struct TypeOfMigratorPass: public ASTMigratorPass,
 
     UnqualifiedLookup Lookup {
       { SF->getASTContext().getIdentifier("type") },
-      SF->getModuleScopeContext(),
-      /*TypeResolver=*/nullptr,
+      ContextStack.empty() ? SF->getModuleScopeContext() : ContextStack.back(),
```
Comment (jrose-apple) [inline lib/Migrator/TypeOfMigratorPass.cpp]: Nitpick: SourceFiles are already module-scope contexts.

**Pull request merged**